astropy/modeling/bounding_box.py

```diff
@@ -680,7 +680,9 @@ def __getitem__(self, key):
         else:
             return self._intervals[self._get_index(key)]
 
-    def bounding_box(self, order: str | None = None):
+    def bounding_box(
+        self, order: str | None = None
+    ) -> tuple[float, float] | tuple[tuple[float, float], ...]:
```

Comment:
In my experience, unions in return type annotations are sometimes worse than no annotation, because IDEs and type checkers will insist that all cases are matched in user code, even when it doesn't make sense. Here however I don't immediately see how to avoid it, as the return type doesn't depend on the arguments (excluding `TypeVar` as a solution), but on the internal state of the object (the `len` of `self._intervals`). If anyone knows any way to express that in type annotations, this would be useful here.